diff --git a/node_modules/@storybook/core-events/dist/cjs/index.js b/node_modules/@storybook/core-events/dist/cjs/index.js
index 68f2828..ff8af71 100644
--- a/node_modules/@storybook/core-events/dist/cjs/index.js
+++ b/node_modules/@storybook/core-events/dist/cjs/index.js
@@ -3,7 +3,7 @@
 Object.defineProperty(exports, "__esModule", {
   value: true
 });
-exports.NAVIGATE_URL = exports.SHARED_STATE_SET = exports.SHARED_STATE_CHANGED = exports.DOCS_RENDERED = exports.STORIES_EXPAND_ALL = exports.STORIES_COLLAPSE_ALL = exports.SELECT_STORY = exports.PREVIEW_KEYDOWN = exports.REGISTER_SUBSCRIPTION = exports.GLOBALS_UPDATED = exports.UPDATE_GLOBALS = exports.RESET_STORY_ARGS = exports.STORY_ARGS_UPDATED = exports.UPDATE_STORY_ARGS = exports.STORY_THREW_EXCEPTION = exports.STORY_ERRORED = exports.STORY_MISSING = exports.STORY_RENDERED = exports.STORY_UNCHANGED = exports.STORY_CHANGED = exports.FORCE_RE_RENDER = exports.CURRENT_STORY_WAS_SET = exports.SET_CURRENT_STORY = exports.SET_STORIES = exports.STORY_SPECIFIED = exports.CHANNEL_CREATED = exports.default = void 0;
+exports.IGNORED_EXCEPTION = exports.NAVIGATE_URL = exports.SHARED_STATE_SET = exports.SHARED_STATE_CHANGED = exports.DOCS_RENDERED = exports.STORIES_EXPAND_ALL = exports.STORIES_COLLAPSE_ALL = exports.SELECT_STORY = exports.PREVIEW_KEYDOWN = exports.REGISTER_SUBSCRIPTION = exports.GLOBALS_UPDATED = exports.UPDATE_GLOBALS = exports.RESET_STORY_ARGS = exports.STORY_ARGS_UPDATED = exports.UPDATE_STORY_ARGS = exports.STORY_THREW_EXCEPTION = exports.STORY_ERRORED = exports.STORY_MISSING = exports.STORY_RENDERED = exports.STORY_UNCHANGED = exports.STORY_CHANGED = exports.FORCE_RE_RENDER = exports.CURRENT_STORY_WAS_SET = exports.SET_CURRENT_STORY = exports.SET_STORIES = exports.STORY_SPECIFIED = exports.CHANNEL_CREATED = exports.default = void 0;
 var events; // Enables: `import Events from ...`
 
 (function (events) {
@@ -33,6 +33,7 @@ var events; // Enables: `import Events from ...`
   events["SHARED_STATE_CHANGED"] = "sharedStateChanged";
   events["SHARED_STATE_SET"] = "sharedStateSet";
   events["NAVIGATE_URL"] = "navigateUrl";
+  events["IGNORED_EXCEPTION"] = "ignoredException";
 })(events || (events = {}));
 
 var _default = events; // Enables: `import * as Events from ...` or `import { CHANNEL_CREATED } as Events from ...`
@@ -64,7 +65,9 @@ var CHANNEL_CREATED = events.CHANNEL_CREATED,
     DOCS_RENDERED = events.DOCS_RENDERED,
     SHARED_STATE_CHANGED = events.SHARED_STATE_CHANGED,
     SHARED_STATE_SET = events.SHARED_STATE_SET,
-    NAVIGATE_URL = events.NAVIGATE_URL;
+    NAVIGATE_URL = events.NAVIGATE_URL,
+    IGNORED_EXCEPTION = events.IGNORED_EXCEPTION;
+exports.IGNORED_EXCEPTION = IGNORED_EXCEPTION;
 exports.NAVIGATE_URL = NAVIGATE_URL;
 exports.SHARED_STATE_SET = SHARED_STATE_SET;
 exports.SHARED_STATE_CHANGED = SHARED_STATE_CHANGED;
diff --git a/node_modules/@storybook/core-events/dist/esm/index.js b/node_modules/@storybook/core-events/dist/esm/index.js
index eb865c0..5142add 100644
--- a/node_modules/@storybook/core-events/dist/esm/index.js
+++ b/node_modules/@storybook/core-events/dist/esm/index.js
@@ -27,6 +27,7 @@ var events; // Enables: `import Events from ...`
   events["SHARED_STATE_CHANGED"] = "sharedStateChanged";
   events["SHARED_STATE_SET"] = "sharedStateSet";
   events["NAVIGATE_URL"] = "navigateUrl";
+  events["IGNORED_EXCEPTION"] = "ignoredException";
 })(events || (events = {}));
 
 export default events; // Enables: `import * as Events from ...` or `import { CHANNEL_CREATED } as Events from ...`
@@ -57,5 +58,6 @@ var CHANNEL_CREATED = events.CHANNEL_CREATED,
     DOCS_RENDERED = events.DOCS_RENDERED,
     SHARED_STATE_CHANGED = events.SHARED_STATE_CHANGED,
     SHARED_STATE_SET = events.SHARED_STATE_SET,
-    NAVIGATE_URL = events.NAVIGATE_URL;
-export { CHANNEL_CREATED, STORY_SPECIFIED, SET_STORIES, SET_CURRENT_STORY, CURRENT_STORY_WAS_SET, FORCE_RE_RENDER, STORY_CHANGED, STORY_UNCHANGED, STORY_RENDERED, STORY_MISSING, STORY_ERRORED, STORY_THREW_EXCEPTION, UPDATE_STORY_ARGS, STORY_ARGS_UPDATED, RESET_STORY_ARGS, UPDATE_GLOBALS, GLOBALS_UPDATED, REGISTER_SUBSCRIPTION, PREVIEW_KEYDOWN, SELECT_STORY, STORIES_COLLAPSE_ALL, STORIES_EXPAND_ALL, DOCS_RENDERED, SHARED_STATE_CHANGED, SHARED_STATE_SET, NAVIGATE_URL };
\ No newline at end of file
+    NAVIGATE_URL = events.NAVIGATE_URL,
+    IGNORED_EXCEPTION = events.IGNORED_EXCEPTION;
+export { CHANNEL_CREATED, STORY_SPECIFIED, SET_STORIES, SET_CURRENT_STORY, CURRENT_STORY_WAS_SET, FORCE_RE_RENDER, STORY_CHANGED, STORY_UNCHANGED, STORY_RENDERED, STORY_MISSING, STORY_ERRORED, STORY_THREW_EXCEPTION, UPDATE_STORY_ARGS, STORY_ARGS_UPDATED, RESET_STORY_ARGS, UPDATE_GLOBALS, GLOBALS_UPDATED, REGISTER_SUBSCRIPTION, PREVIEW_KEYDOWN, SELECT_STORY, STORIES_COLLAPSE_ALL, STORIES_EXPAND_ALL, DOCS_RENDERED, SHARED_STATE_CHANGED, SHARED_STATE_SET, NAVIGATE_URL, IGNORED_EXCEPTION };
\ No newline at end of file
diff --git a/node_modules/@storybook/core-events/dist/modern/index.js b/node_modules/@storybook/core-events/dist/modern/index.js
index 395c356..ccfafc0 100644
--- a/node_modules/@storybook/core-events/dist/modern/index.js
+++ b/node_modules/@storybook/core-events/dist/modern/index.js
@@ -27,6 +27,7 @@ var events; // Enables: `import Events from ...`
   events["SHARED_STATE_CHANGED"] = "sharedStateChanged";
   events["SHARED_STATE_SET"] = "sharedStateSet";
   events["NAVIGATE_URL"] = "navigateUrl";
+  events["IGNORED_EXCEPTION"] = "ignoredException";
 })(events || (events = {}));
 
 export default events; // Enables: `import * as Events from ...` or `import { CHANNEL_CREATED } as Events from ...`
@@ -58,5 +59,6 @@ export const {
   DOCS_RENDERED,
   SHARED_STATE_CHANGED,
   SHARED_STATE_SET,
-  NAVIGATE_URL
+  NAVIGATE_URL,
+  IGNORED_EXCEPTION
 } = events;
\ No newline at end of file
diff --git a/node_modules/@storybook/core-events/dist/ts3.4/index.d.ts b/node_modules/@storybook/core-events/dist/ts3.4/index.d.ts
index fc3f3bc..6114e81 100644
--- a/node_modules/@storybook/core-events/dist/ts3.4/index.d.ts
+++ b/node_modules/@storybook/core-events/dist/ts3.4/index.d.ts
@@ -24,7 +24,8 @@ declare enum events {
     DOCS_RENDERED = "docsRendered",
     SHARED_STATE_CHANGED = "sharedStateChanged",
     SHARED_STATE_SET = "sharedStateSet",
-    NAVIGATE_URL = "navigateUrl"
+    NAVIGATE_URL = "navigateUrl",
+    IGNORED_EXCEPTION = "ignoredException"
 }
 export default events;
-export declare const CHANNEL_CREATED: events, STORY_SPECIFIED: events, SET_STORIES: events, SET_CURRENT_STORY: events, CURRENT_STORY_WAS_SET: events, FORCE_RE_RENDER: events, STORY_CHANGED: events, STORY_UNCHANGED: events, STORY_RENDERED: events, STORY_MISSING: events, STORY_ERRORED: events, STORY_THREW_EXCEPTION: events, UPDATE_STORY_ARGS: events, STORY_ARGS_UPDATED: events, RESET_STORY_ARGS: events, UPDATE_GLOBALS: events, GLOBALS_UPDATED: events, REGISTER_SUBSCRIPTION: events, PREVIEW_KEYDOWN: events, SELECT_STORY: events, STORIES_COLLAPSE_ALL: events, STORIES_EXPAND_ALL: events, DOCS_RENDERED: events, SHARED_STATE_CHANGED: events, SHARED_STATE_SET: events, NAVIGATE_URL: events;
+export declare const CHANNEL_CREATED: events, STORY_SPECIFIED: events, SET_STORIES: events, SET_CURRENT_STORY: events, CURRENT_STORY_WAS_SET: events, FORCE_RE_RENDER: events, STORY_CHANGED: events, STORY_UNCHANGED: events, STORY_RENDERED: events, STORY_MISSING: events, STORY_ERRORED: events, STORY_THREW_EXCEPTION: events, UPDATE_STORY_ARGS: events, STORY_ARGS_UPDATED: events, RESET_STORY_ARGS: events, UPDATE_GLOBALS: events, GLOBALS_UPDATED: events, REGISTER_SUBSCRIPTION: events, PREVIEW_KEYDOWN: events, SELECT_STORY: events, STORIES_COLLAPSE_ALL: events, STORIES_EXPAND_ALL: events, DOCS_RENDERED: events, SHARED_STATE_CHANGED: events, SHARED_STATE_SET: events, NAVIGATE_URL: events, IGNORED_EXCEPTION: events;
diff --git a/node_modules/@storybook/core-events/dist/ts3.9/index.d.ts b/node_modules/@storybook/core-events/dist/ts3.9/index.d.ts
index 25b4c1d..e3cb035 100644
--- a/node_modules/@storybook/core-events/dist/ts3.9/index.d.ts
+++ b/node_modules/@storybook/core-events/dist/ts3.9/index.d.ts
@@ -24,7 +24,8 @@ declare enum events {
     DOCS_RENDERED = "docsRendered",
     SHARED_STATE_CHANGED = "sharedStateChanged",
     SHARED_STATE_SET = "sharedStateSet",
-    NAVIGATE_URL = "navigateUrl"
+    NAVIGATE_URL = "navigateUrl",
+    IGNORED_EXCEPTION = "ignoredException"
 }
 export default events;
-export declare const CHANNEL_CREATED: events, STORY_SPECIFIED: events, SET_STORIES: events, SET_CURRENT_STORY: events, CURRENT_STORY_WAS_SET: events, FORCE_RE_RENDER: events, STORY_CHANGED: events, STORY_UNCHANGED: events, STORY_RENDERED: events, STORY_MISSING: events, STORY_ERRORED: events, STORY_THREW_EXCEPTION: events, UPDATE_STORY_ARGS: events, STORY_ARGS_UPDATED: events, RESET_STORY_ARGS: events, UPDATE_GLOBALS: events, GLOBALS_UPDATED: events, REGISTER_SUBSCRIPTION: events, PREVIEW_KEYDOWN: events, SELECT_STORY: events, STORIES_COLLAPSE_ALL: events, STORIES_EXPAND_ALL: events, DOCS_RENDERED: events, SHARED_STATE_CHANGED: events, SHARED_STATE_SET: events, NAVIGATE_URL: events;
+export declare const CHANNEL_CREATED: events, STORY_SPECIFIED: events, SET_STORIES: events, SET_CURRENT_STORY: events, CURRENT_STORY_WAS_SET: events, FORCE_RE_RENDER: events, STORY_CHANGED: events, STORY_UNCHANGED: events, STORY_RENDERED: events, STORY_MISSING: events, STORY_ERRORED: events, STORY_THREW_EXCEPTION: events, UPDATE_STORY_ARGS: events, STORY_ARGS_UPDATED: events, RESET_STORY_ARGS: events, UPDATE_GLOBALS: events, GLOBALS_UPDATED: events, REGISTER_SUBSCRIPTION: events, PREVIEW_KEYDOWN: events, SELECT_STORY: events, STORIES_COLLAPSE_ALL: events, STORIES_EXPAND_ALL: events, DOCS_RENDERED: events, SHARED_STATE_CHANGED: events, SHARED_STATE_SET: events, NAVIGATE_URL: events, IGNORED_EXCEPTION: events;
