diff --git a/node_modules/@storybook/client-api/dist/modern/story_store.js b/node_modules/@storybook/client-api/dist/modern/story_store.js
index d1e0e21..2b3f5ae 100644
--- a/node_modules/@storybook/client-api/dist/modern/story_store.js
+++ b/node_modules/@storybook/client-api/dist/modern/story_store.js
@@ -599,7 +599,7 @@ export default class StoryStore {
 
     const initialArgs = this._argsEnhancers.reduce((accumulatedArgs, enhancer) => Object.assign({}, accumulatedArgs, enhancer(Object.assign({}, identification, {
       parameters: combinedParameters,
-      args: initialArgsBeforeEnhancers,
+      args: accumulatedArgs,
       argTypes,
       globals: {},
       originalStoryFn: getOriginal()
