diff --git a/node_modules/@storybook/client-api/dist/modern/story_store.js b/node_modules/@storybook/client-api/dist/modern/story_store.js
index d1e0e21..1f13684 100644
--- a/node_modules/@storybook/client-api/dist/modern/story_store.js
+++ b/node_modules/@storybook/client-api/dist/modern/story_store.js
@@ -606,10 +606,15 @@ export default class StoryStore {
     }))), initialArgsBeforeEnhancers);
 
     const runPlayFunction = async () => {
-      const {
-        play
-      } = combinedParameters;
-      return play ? play() : undefined;
+      const { play } = combinedParameters;
+      if (!play) return;
+      try {
+        const { viewMode, storyId } = this._selection;
+        const containerId = viewMode === "docs" ? `story--${storyId}` : 'root'
+        play({ viewMode, storyId, containerId })
+      } catch (e) {
+        if (typeof e !== Symbol) throw e;
+      }
     };
 
     _stories[id] = Object.assign({}, identification, {
